### PR TITLE
[IOTDB-6083] Pipe: Fix subscrption running with the pattern option causing OOM & make PipeRawTabletInsertionEvent able to report progress to avoid losing data

### PR DESCRIPTION
**Pull request by DanielWang2035**
This PR will fix the issue in [Pipe: Avoid subscrption running with the pattern option causing OOM](https://github.com/apache/iotdb/pull/10767)

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/row/PipeRowCollector.java [hunk 1]
```diff
@@ -85,7 +94,8 @@ public void collectRow(Row row) {
 
   private void collectTabletInsertionEvent() {
     if (tablet != null) {
-      tabletInsertionEventList.add(new PipeRawTabletInsertionEvent(tablet, isAligned));
+      tabletInsertionEventList.add(
+          new PipeRawTabletInsertionEvent(tablet, isAligned, pipeTaskMeta, sourceEvent, true));
```
Comment (SteveYurongSu) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/row/PipeRowCollector.java:98]: can not always be true...

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/async/IoTDBThriftAsyncConnector.java [hunk 2]
```diff
@@ -157,6 +157,21 @@ public void transfer(TabletInsertionEvent tabletInsertionEvent) throws Exception
       return;
     }
 
+    if (((EnrichedEvent) tabletInsertionEvent).shouldParsePattern()) {
+      for (final TabletInsertionEvent event :
+          tabletInsertionEvent.processRowByRow(
```
Comment (SteveYurongSu) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/async/IoTDBThriftAsyncConnector.java:162]: What about other connectors? AirGapConnector?

Comment (SteveYurongSu) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/async/IoTDBThriftAsyncConnector.java:162]: There can be a more efficient way... Please refer to PipeRawTabletInsertionEvent#convertToTablet

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/row/PipeRowCollector.java (hunk 1 above)
Comment (DanielWang2035) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/row/PipeRowCollector.java]: fixed

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/async/IoTDBThriftAsyncConnector.java (hunk 2 above)
Comment (DanielWang2035) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/async/IoTDBThriftAsyncConnector.java:162]: fixed for AirGapConnector

Comment (DanielWang2035) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/async/IoTDBThriftAsyncConnector.java]: changed

**Review by SteveYurongSu — APPROVED** [review]

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**